### PR TITLE
feat: sort ignore case

### PR DESCRIPTION
**Pull request by Linus789**

### CODE REVIEW

**Comment by sxyazi** [comment]
Thank you for the PR!

Which do you think is more common, case sensitive and case insensitive? i.e. is it necessary to set case insensitivity as the default?

**Comment by Linus789** [comment]
I can only speak for myself, but I prefer it to be case insensitive. And what default do you mean? The yazi.toml one, or the ones in the keymap?

**Comment by sxyazi** [comment]
Yep, I agree with that as well. It seems that sorting by ASCII ordering doesn't make much sense, and case insensitivity makes more sense.

> The yazi.toml one, or the ones in the keymap?

Both.

File: core/src/files/sorter.rs [hunk 1]
```diff
@@ -32,7 +32,7 @@ impl FilesSorter {
 
 		match self.by {
 			SortBy::Alphabetical => items.sort_unstable_by(|a, b| {
-				if self.ignore_case {
+				if self.sensitive {
```
Comment (Linus789) [inline core/src/files/sorter.rs:35]: Shouldn't it be the inverse?

Comment (sxyazi) [reply core/src/files/sorter.rs:35]: Yep, thank the reminder!

File: core/src/files/sorter.rs [hunk 2]
```diff
@@ -65,12 +75,16 @@ impl FilesSorter {
 		indices.sort_unstable_by(|&a, &b| {
 			let promote = self.promote(entities[a].1, entities[b].1);
 			if promote != Ordering::Equal {
-				promote
-			} else if self.reverse {
-				natord::compare(&entities[b].0, &entities[a].0)
+				return promote;
+			}
+
+			let ordering = if self.sensitive {
```
Comment (Linus789) [inline core/src/files/sorter.rs:81]: Here too

**Comment by sxyazi** [comment]
Thank you, merged!

**Pull request merged**